library/ssl_tls12_server.c

```diff
@@ -763,7 +764,12 @@ static int ssl_pick_cert(mbedtls_ssl_context *ssl,
         }
 
 #if defined(MBEDTLS_KEY_EXCHANGE_ECDSA_CERT_REQ_ALLOWED_ENABLED)
-        if (pk_alg == MBEDTLS_PK_ECDSA &&
```

Comment:
This looks like a preexisting bug: we were checking `pk_alg == MBEDTLS_PK_ECDSA`, but when `MBEDTLS_USE_PSA_CRYPTO` is enabled, the variable `pk_alg` contained a `psa_algorithm_t` value, not an `mbedtls_pk_type_t` value. This couldn't have matched when it shouldn't, but it didn't match when it should. So under `MBEDTLS_USE_PSA_CRYPTO`, the call to `ssl_check_key_curve` was always skipped. I don't know what the consequences of this are.